test/tests/functional/pbs_cray_suspend_resume.py

```diff
@@ -93,3 +97,390 @@ def test_exclusive_job_not_suspended(self):
+
+        try:
+            self.server.sigjob(jid1, "admin-resume", runas=ROOT_USER)
+        except PbsSignalError as e:
+            self.assertTrue(
+                'Job can not be resumed with the requested resume signal'
+                in e.msg[0])
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+
+        j2 = Job(TEST_USER)
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'R', ATTR_substate: 42}, id=jid2)
+        self.server.sigjob(jid2, "admin-suspend", runas=ROOT_USER)
+        self.server.expect(JOB, {ATTR_state: 'S', ATTR_substate: 43}, id=jid2)
+
+        try:
+            self.server.sigjob(jid2, "resume", runas=ROOT_USER)
+        except PbsSignalError as e:
+            self.assertTrue(
+                'Job can not be resumed with the requested resume signal'
+                in e.msg[0])
+
+        # If resume had worked, the job would be in substate 45
+        self.server.expect(JOB, {ATTR_state: 'S', ATTR_substate: 43}, id=jid2)
+        self.server.cleanup_jobs()
+
+    def submit_resv(self, resv_start, chunks, resv_dur):
+        a = {'Resource_List.select': '%d:ncpus=1:vntype=cray_compute' % chunks,
+             'Resource_List.place': 'scatter',
+             'reserve_start': int(resv_start),
+             'reserve_duration': int(resv_dur)
+             }
+        r = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(r)
+        try:
+            a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+            d = self.server.expect(RESV, a, id=rid)
+        except PtlExpectError, e:
+            d = e.rv
+        return d
+
+    def test_preempt_STF(self):
+        """
+        Test shrink to fit by creating one reservation having ncpus=1,
+        starting in 3 min. with a duration of two hours.  A preempted STF job
+        with min_walltime of 2 min. and max_walltime of 2 hours will stay
+        suspended after higher priority job goes away if its
+        min_walltime can't be satisfied.
+        """
+        qname = 'highp'
+        queues = self.server.status(QUEUE)
+        for q in queues:
+            if 'id' in q and q['id'] == qname:
+                rc = self.server.manager(MGR_CMD_DELETE, QUEUE, None, qname)
+                self.assertEqual(rc, 0)
+                break
+        a = {'queue_type': 'execution'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, qname)
+        a = {'enabled': 'True', 'started': 'True', 'priority': '150'}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, qname)
+
+        # Reserve all the compute nodes
+        vlist = []
+        try:
+            vnl = self.server.filter(
+                VNODE, {'resources_available.vntype': 'cray_compute'})
+            vlist = vnl["resources_available.vntype=cray_compute"]
+        except Exception:
+            sys.exc_clear()
+            self.logger.info("There are no cray_compute vnodes present.")
```

Comment:
Shouldn't we be asserting from here because if there are no vnodes, the test can not run

Reply:
It is good python style to look for a smaller set of exceptions than the blanket Exception (which is better than just except:).  What exceptions are you expecting?  A ValueError?  Look for that.